code/cow_brd.c

```diff
@@ -138,6 +140,19 @@ static struct page *brd_insert_page(struct brd_device *brd, sector_t sector)
 
   radix_tree_preload_end();
 
+  // Copy over the data in the parent's page if it exists.
+  if (brd->parent_brd) {
+    parent_page = brd_lookup_page(brd->parent_brd, sector);
+    // This page may not have originally existed in the parent.
+    if (parent_page) {
+      dst = kmap_atomic(page);
+      parent_src = kmap_atomic(parent_page);
+      memcpy(dst, parent_src, PAGE_SIZE);
+      kunmap_atomic(parent_src);
+      kunmap_atomic(dst);
```

Comment:
Not sure what I get whats going on here. We copy over the page and then unmap it?

Reply:
It's just setup, it copies over the data that was in the base disk image for this page. That way the direct IO and regular IO accesses later on don't have to worry about handling that the first time around. If the page didn't exist in the parent's base image then it does not copy over data.